Userland/Shell/Shell.cpp

```diff
@@ -1130,7 +1131,7 @@ bool Shell::is_allowed_to_modify_termios(const AST::Command& command) const
 
 void Shell::restore_ios()
 {
-    if (m_is_subshell)
+    if (m_is_subshell || !m_is_interactive)
```

Comment:
This will break job control for scripts, please only skip the tcsetattr() call for non-interactive sessions.